### PR TITLE
LevelsHelper & LevelsController cleanup

### PR DESCRIPTION
**Pull request by joshlory**
- Fewer `@instance` variables in `LevelsHelper`
- Make `embed_level` its own action
- Move Callout tests to `LevelsHelperTest` so they no longer rely on `assigns(:callouts_to_show)`

### CODE REVIEW

**Comment by wjordan** [comment]
The refactoring on complicated methods and code style cleanups look very nice! Removing those side-effects from that callouts code path will make it much easier to reason about and maintain.

However, I'm not sure I understand the reasoning behind moving the `embed_blocks` and `embed_level` actions into the ScriptLevelsController, when its actions are not at all associated with the ScriptLevel model. It seems like this move ends up adding more special-case configuration than it simplifies.

File: dashboard/app/controllers/levels_controller.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ class LevelsController < ApplicationController
   before_filter :authenticate_user!, :except => [:embed_blocks], :unless => Proc.new {params[:embed] && action_name == 'show'}
```
Comment (wjordan) [inline dashboard/app/controllers/levels_controller.rb]: These exceptions should also no longer be needed if these actions are moved from this controller.

**Comment by joshlory** [comment]
Will, totally agree.  What do you think about a new EmbedController that hosts these two methods?  That way LevelsController only contains admin actions (maybe rename to AdminLevelsController?).  I don't know if it deviates from The Rails Way™ to have more controllers than models.  But if we think of controllers as collections of actions for a given resource, two separate groupings may make more sense.

**Comment by wjordan** [comment]
Laurel would probably know more definitively, but I do agree to think of a controller as a collection of actions for a given resource - especially since the resourceful-route (`resources`) makes that relation quite explicit.

I think adding an `EmbedController` would suggest that there are `embed` resources that both of these actions act upon, which isn't the case here (there's no distinct ID that refers to an `embed` resource aside from the existing level id). They're more like two customized views into `level` resources. A ScriptLevel is also a kind of view into `level`, making that controller a bit muddled, but it's also a distinct resource with an ID that uniquely identifies it.

I think the underlying confusion stems from the desire to keep `LevelsController` restricted to admins. As long as `embed_level` and `embed_blocks` are reading a `level` directly and not operating on a different resource, I think it makes sense to relax the admin-only restriction on the `level`'s controller. We could discuss whether or not reading a `level` directly is a problem for other reasons (e.g., the fact that level ids are not consistent across environments may be the reason we want to prevent read access to `level` resources by non-admins, in which case we might want to rethink the design of these embedded views altogether). But as they are currently designed, I still think the place that makes the most sense for these embed_* actions to live is in LevelsController.

**Comment by joshlory** [comment]
Yep that's fair.  I've moved these two actions back to `LevelsController`.

**Comment by wjordan** [comment]
:+1:
Nice work moving the embed_level into its own controller action, that makes what's going on there much more readable.
This refactor will be really helpful!

**Pull request merged**